src/abstraction/transient.rs

```diff
@@ -99,7 +102,67 @@ impl TransientKeyContext {
         Ok((key_context, key_auth))
     }
 
-    fn get_public_from_params(&self, params: KeyParams) -> Result<TPM2B_PUBLIC> {
+    /// Create a new signing, encryption and decryption key.
+    ///
+    /// A key is created as a descendant of the context root key, with the given parameters.
+    ///
+    /// If successful, the result contains the saved context of the key and a vector of
+    /// bytes forming the authentication value for said key.
+    ///
+    /// # Constraints
+    /// * `auth_size` must be at most 32
+    ///
+    /// # Errors
+    /// * if the authentication size is larger than 32 a `WrongParamSize` wrapper error is returned
+    /// * for RSA keys, if the specified key size is not one of 1024, 2048, 3072 or 4096, `WrongParamSize`
+    /// is returned
+    /// * if the key_params is not for an RSA key, `InvalidParam` is returned
+    /// * if the key_params does not have an `AnySig` scheme, `InvalidParam` is returned
+    /// * errors are returned if any method calls return an error: `Context::get_random`,
+    /// `TransientKeyContext::set_session_attrs`, `Context::create_key`, `Context::load`,
+    /// `Context::context_save`, `Context::context_flush`
+    pub fn create_signing_encryption_decryption_key(
+        &mut self,
+        key_params: KeyParams,
+        auth_size: usize,
+    ) -> Result<(TpmsContext, Option<Auth>)> {
+        if auth_size > 32 {
+            return Err(Error::local_error(ErrorKind::WrongParamSize));
+        }
+        let key_auth = if auth_size > 0 {
+            self.set_session_attrs()?;
+            let random_bytes = self.context.get_random(auth_size)?;
+            Some(Auth::try_from(random_bytes.value().to_vec())?)
+        } else {
+            None
+        };
+
+        self.set_session_attrs()?;
+        let (key_priv, key_pub) = self.context.create_key(
+            self.root_key_handle,
+            &self.get_public_from_params(key_params, true)?,
+            key_auth.as_ref(),
+            None,
+            None,
+            &[],
+        )?;
+        self.set_session_attrs()?;
+        let key_handle = self.context.load(self.root_key_handle, key_priv, key_pub)?;
+
+        self.set_session_attrs()?;
+        let key_context = self.context.context_save(key_handle).or_else(|e| {
+            self.context.flush_context(key_handle)?;
+            Err(e)
+        })?;
+        self.context.flush_context(key_handle)?;
+        Ok((key_context, key_auth))
+    }
+
+    fn get_public_from_params(
+        &self,
+        params: KeyParams,
+        encryption_decryption: bool,
+    ) -> Result<TPM2B_PUBLIC> {
```

Comment:
I think instead of using a `bool` to signal which path to take you can change the `KeyParams` enum to have `RsaSign` and `RsaEncrypt` variants, and for the latter you don't allow a scheme, just size and public exponent.